pkg/monitors/http/http.go

```diff
@@ -0,0 +1,254 @@
+	}
+	if _, err := auth.TLSConfig(tlsCfg, m.conf.CACertPath, m.conf.ClientCertPath, m.conf.ClientKeyPath); err != nil {
+		return nil, err
+	}
+	conn := tls.Client(ipConn, tlsCfg)
+	if err != nil {
+		return
+	}
+	defer conn.Close()
+	err = conn.Handshake()
+	if err != nil {
+		logger.WithError(err).Error("failed during handshake")
+		valid = false
+	}
+	certs := conn.ConnectionState().PeerCertificates
+	for i, cert := range certs {
+		opts := x509.VerifyOptions{
+			Intermediates: x509.NewCertPool(),
+		}
+		if i == 0 {
+			opts.DNSName = host
+			for j, cert := range certs {
+				if j != 0 {
+					opts.Intermediates.AddCert(cert)
+				}
+			}
+			secondsLeft = time.Until(cert.NotAfter).Seconds()
+		}
+		_, err := cert.Verify(opts)
+		if err != nil {
+			logger.WithError(err).Info("failed verify certificate")
+			valid = false
+		}
+	}
+	dimensions := map[string]string{
+		"url":        site.String(),
+		"serverName": host,
+		"isValid":    strconv.FormatBool(valid),
+	}
+	dps = append(dps, datapoint.New(httpCertExpiry, dimensions, datapoint.NewFloatValue(secondsLeft), datapoint.Gauge, time.Time{}))
+	return dps, nil
+}
+
+func (m *Monitor) getHTTPStats(site string, logger *logrus.Entry) (dps []*datapoint.Datapoint, lastURL string, err error) {
+	// Init http client
+	client, err := m.conf.HTTPConfig.Build()
+	if err != nil {
+		return
+	}
+	// Init body if applicable
+	var body io.Reader
+	if m.conf.Body != "" {
+		body = strings.NewReader(m.conf.Body)
+	}
+	if m.conf.NoRedirects {
+		logger.Debug("Do not follow redirects")
+		client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		}
+	}
+	req, err := http.NewRequest(m.conf.Method, site, body)
+	if err != nil {
+		return
+	}
+	// starts timer
+	now := time.Now()
+	resp, err := client.Do(req)
+	if err != nil {
+		return
+	}
+	responseTime := time.Since(now).Seconds()
+	defer resp.Body.Close()
+	lastURL = resp.Request.URL.String()
+	dimensions := map[string]string{
+		"url":    lastURL,
+		"method": m.conf.Method,
+	}
+	matchCode := false
+	statusCode := resp.StatusCode
+	if statusCode == m.conf.DesiredCode {
+		matchCode = true
+	}
+	dps = append(dps,
+		datapoint.New(httpResponseTime, dimensions, datapoint.NewFloatValue(responseTime), datapoint.Gauge, time.Time{}),
+		datapoint.New(httpStatusCode, addDimensions(dimensions, map[string]string{
+			"matchCode":   strconv.FormatBool(matchCode),
+			"desiredCode": strconv.Itoa(m.conf.DesiredCode),
+		}), datapoint.NewIntValue(int64(statusCode)), datapoint.Gauge, time.Time{}),
+	)
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.WithError(err).Error("could not parse body response")
+	} else {
+		contentDimensions := dimensions
+		if m.conf.Regex != "" {
+			matchRegex := false
+			if m.regex.Match(bodyBytes) {
+				matchRegex = true
+			}
+			contentDimensions = addDimensions(contentDimensions, map[string]string{
+				"matchRegex": strconv.FormatBool(matchRegex),
```

Comment:
Same issue as for `isValid` with TLS above.  There should be a separate metric called something like`http.content_matched`.